holonix/test/hc-sandbox.bats

```diff
@@ -0,0 +1,53 @@
+
+#!/usr/bin/env bats
+
+setup() {
+  BATS_TMPDIR="$(mktemp -d)"
+  cd "${BATS_TMPDIR:?}"
+}
+
+teardown() {
+  cd ..
+  rm -rf "${BATS_TMPDIR:?}"
+}
+
+@test "expected hc-sandbox to be available" {
+  result="$(type -f hc-sandbox)"
+  echo $result
+  [[ "$result" == *"hc-sandbox" ]]
+}
+
+@test "expected hc-sandbox to clean, create, and run a sandbox" {
+  set -muEeo pipefail
+
+  echo pass | hc-sandbox --piped create -n1
+  (echo pass | hc-sandbox --piped run 0) &
+
+  declare result
+  result=0
+
+  # succedd if we can reach the admin interface within 10 seconds
+  for i in `seq 1 10`; do
+    sleep 1
+    (
+      set -eE
```

Comment:
Both already set above?

Reply:
i've experienced cases where `( ... )` didn't inherit these settings from their parent shell and errors in subshells didn't cause failure of the parent script

Reply:
```suggestion
```

double-checked the manual and it's not required to re-specify because `-E` is used in the parent shell:

>               -E      If set, any trap on ERR is inherited by shell functions, command substitutions, and commands executed in a subshell environment.  The ERR trap is normally not inherited in such cases.